ci/container_scripts/install_deps.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+# Dependencies that depend only on CONTAINER should be installed here.
+
+set -euo pipefail
+
+APT_PACKAGES="
+  cmake
+  curl
+  findutils
+  libc-dbg
+  libglib2.0-0
+  libglib2.0-dev
+  libigraph0-dev
+  libigraph0v5
+  libprocps-dev
+  make
+  python3
+  xz-utils
+  "
+
+RPM_PACKAGES="
+  cmake
+  curl
+  findutils
+  glib2
+  glib2-devel
+  igraph
+  igraph-devel
+  make
+  procps-devel
+  python3
+  xz
+  xz-devel
+  yum-utils
+  "
+
+case "$CONTAINER" in
+    ubuntu:*|debian:*)
+        sed -i '/deb-src/s/^# //' /etc/apt/sources.list
+        DEBIAN_FRONTEND=noninteractive apt-get update
+        DEBIAN_FRONTEND=noninteractive apt-get install -y $APT_PACKAGES
+        ;;
```

Comment:
At the end of each case (in this file and install_extra_deps), I think it would be good to clear the package repository cache (with `rm -rf /var/lib/apt/lists/*` or `dnf clean all`) to avoid saving stale repository information into the layers. This will add ~5-10 seconds to the build time, but I think is worthwhile to avoid problems where we rebuild an image from Docker's cached layers.

Reply:
So a possible problem is you build the image, and in step `install_deps` the package manager creates a package cache, uses it, and then uses that same cache in the `install_extra_deps`. This is fine (other than the increased disk space in the image and its layers) until you edit something in the `install_extra_deps` step which will rebuild the image starting from that step. Now for apt will continue using the package cache saved into the image from the previous step, but this cache may be very old and `apt install` for example will fail. Usually it's recommended to delete the package cache at the end of each 'run' step, and re-download it at the start of the next step if needed. This might explain it better: https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run

Reply:
I see what you mean, but I'm still leaning towards the extra overhead of downloading and deleting the package cache twice being worth it. Having to remember to use `-n` / `--no-cache` after adding a new dependency doesn't seem too bad, and is hopefully a fairly obvious solution when there are errors about the package not existing.